File: AMS_基础信息组件/code/server/source/com.yss.ams.base.information.support/src/main/java/com/yss/ams/base/information/support/sys/convertdict/zdorg/service/IZdOrgService.java
package com.yss.ams.base.information.support.sys.convertdict.zdorg.service;

import java.util.HashMap;

import com.yss.framework.api.mvc.biz.IServiceBus;
import com.yss.framework.api.restful.annotations.RestfulSupported;
import com.yss.framework.api.servlet.pojo.QueryRes;

/**
 * 机构转换字典普通服务接口，主要进行增删改查操作
 *
 */
@RestfulSupported
public interface IZdOrgService extends IServiceBus{

	/**
	 * 查询A区数据方法定义
	 * 
	 * @param paraMap
	 * @return
	 */
	public QueryRes getTreeViewData(HashMap<String, Object> paraMap);
}

File: AMS_基础信息组件/code/client/source/YssInformation.Support/Sys/ConvertDict/Zdorg/Service/IZdOrgService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FAST.Common.Service.Pojo;
using FAST.Common.Service.Services.Base;
using FAST.Common.Service.Attributes;


namespace YssInformation.Support.Sys.ConvertDict.Zdorg.Service
{
    [NewServiceAttribute(ServiceType.NEW_SVC)]
    [ServiceAttribute(ServiceId = YssInformation.Support.common.ServiceIdConstant.OSGI_BASEBUSINESS, ServiceController = "com.yss.ams.base.information.support.sys.convertdict.zdorg.controller.IZdOrgController", ServiceUrl = "")]
    public interface IZdOrgService : IServiceBus
    {
        /// <summary>
        /// 获取树型数据结构
        /// </summary>
        /// <param name="paraDict">查询条件参数对象</param>
        /// <returns>查询结果集</returns>
        [MethodAttribute(Method = RequestType.POST, MethodUrl = "/getTreeViewData")]
        QueryRes getTreeViewData(Dictionary<string, string> paraDict);

        /// <summary>
        /// Root
        /// </summary>
        /// <returns>Root</returns>
        [MethodAttribute(Method = RequestType.POST, MethodUrl = "/checkHasRootNode")]
        string checkHasRootNode();

        /// <summary>
        /// 此目录下是否有数据
        /// </summary>
        /// <returns>有：true ;没有：false</returns>
        [MethodAttribute(Method = RequestType.POST, MethodUrl = "/isHaveData")]
        bool isHaveData();
    }
}
